doc/source/cluster/kubernetes/user-guides/ray-k8s-setup.md

```diff
@@ -0,0 +1,81 @@
+(ray-kubernetes-setup)=
+
+# Set up a Ray + Kubernetes cluster
+
+This document contains recommendations for setting up a Ray + Kubernetes cluster for your organization.
+
+When you set up Ray on Kubernetes, the KubeRay documentation provides an overview of how to configure the operator to execute and manage the Ray cluster lifecycle. This guide complements the KubeRay documentation by providing best practices for effectively using Ray deployments in your organization.
+
+This guide covers best practices for these deployment considerations:
+
+* Where to ship or run your code on the Ray cluster
+* Choosing a storage system for artifacts
+* Package dependencies for your application
+
+Deployment considerations are different for development and production. This table summarizes the recommended setup for both interactive development and production:
+
+|   | Interactive Development  | Production  |
+|---|---|---|
+| Cluster Configuration  | KubeRay YAML  | KubeRay YAML  |
+| Code | Run driver or Jupyter notebook on head node | S3 + runtime envs <br /> OR <br /> Bake code into Docker image (link)  |
+| Artifact Storage | Set up an EFS  | Cloud storage (S3, GS)  |
+| Package Dependencies | Install onto NFS <br /> or <br /> Use runtime environments | Bake into docker image  |
+
+Table 1: Table comparing recommended setup for development and production.
+
+## Interactive development
+
+To provide an interactive development environment for data scientists, you should set up the code, storage, and dependencies in a way that reduces context switches for developers and shortens iteration times.
+
+```{eval-rst}
+.. image:: ../images/interact-dev.png
+    :align: center
+..
+    Find the source document here (https://whimsical.com/clusters-P5Y6R23riCuNb6xwXVXN72)
+```
+
+### Storage
+
+Use one of these two standard solutions for artifact and log storage during the development process:
```

Comment:
It is inconsistent with the table above. We only mention NFS/EFS in the table under the 'interactive development' column. However, here we reference both NFS/EFS and S3/GS.

Reply:
updated